management/server/peer.go

```diff
@@ -577,24 +586,30 @@ func (am *DefaultAccountManager) SyncPeer(ctx context.Context, sync PeerSync, ac
 		return nil, nil, nil, err
 	}
 
-	var postureChecks []*posture.Checks
-
 	if peerNotValid {
 		emptyMap := &NetworkMap{
 			Network: account.Network.Copy(),
 		}
-		return peer, emptyMap, postureChecks, nil
+		return peer, emptyMap, nil, nil
+	}
+
+	peer, peerMetaUpdated := updatePeerMeta(peer, sync.Meta, account)
+	if peerMetaUpdated {
+		err = am.Store.SaveAccount(ctx, account)
+		if err != nil {
+			return nil, nil, nil, err
+		}
 	}
 
-	if isStatusChanged {
+	if isStatusChanged || (peerMetaUpdated && sync.UpdateAccountPeers) {
```

Comment:
I wonder what would be a usecase where a metadata update requires a network map update beeing send?